manager/integration/tests/common.py

```diff
@@ -4138,13 +4139,23 @@ def create_and_wait_deployment(apps_api, deployment_manifest):
 
 def wait_and_get_any_deployment_pod(core_api, deployment_name,
                                     is_phase="Running"):
+    stable_pod = None
```

Comment:
Suggest adding a comment to describe why we do this way to checking `WAIT_FOR_POD_STABLE_MAX_RETRY` times to ensure the pod is really ready (still best efforts).

Also, create a github issue for resolving the problem about the pod is somehow sometimes deleted by longhorn manager a few times to cause this test flaky. (ref: https://github.com/longhorn/longhorn/issues/4382#issuecomment-1297639493)

Reply:
issue created (https://github.com/longhorn/longhorn/issues/4814) and comment added.